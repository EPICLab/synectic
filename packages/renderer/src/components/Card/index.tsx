export {default} from './Card';
export {default as BaseCard} from './BaseCard';
