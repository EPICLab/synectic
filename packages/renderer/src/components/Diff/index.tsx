export {default} from './Diff';
export {default as DiffReverse} from './DiffReverse';
