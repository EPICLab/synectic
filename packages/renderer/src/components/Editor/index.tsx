export {default} from './Editor';
export {default as EditorReverse} from './EditorReverse';
