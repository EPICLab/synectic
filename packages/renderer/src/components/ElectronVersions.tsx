import {versions} from '#preload';

const ElectronVersion = () => {
  return (
    <table id="process-versions">
      <tbody>
        {Object.entries(versions).map(([lib, version]) => (
          <tr
            key={lib}
            aria-label="versions-row"
          >
            <td>{lib}</td>
            <td>v{version}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default ElectronVersion;
