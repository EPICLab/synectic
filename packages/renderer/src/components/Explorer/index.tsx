export {default} from './Explorer';
export {default as ExplorerReverse} from './ExplorerReverse';
