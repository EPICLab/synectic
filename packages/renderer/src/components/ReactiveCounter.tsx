import {useState} from 'react';

const ReactiveCounter = () => {
  const [count, setCount] = useState(0);

  const increase = () => setCount(count + 1);

  return <button onClick={increase}> count is: {count}</button>;
};

export default ReactiveCounter;
