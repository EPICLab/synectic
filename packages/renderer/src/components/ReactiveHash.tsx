import {useState} from 'react';
import {sha256sum} from '#preload';

const ReactiveHash = () => {
  const [rawString, setRawString] = useState('Hello World');
  const hashedString = sha256sum(rawString);

  return (
    <table>
      <tbody>
        <tr>
          <th>
            <label htmlFor="reactive-hash-raw-value">Raw value :</label>
          </th>
          <td>
            <input
              type="text"
              id="reactive-hash-raw-value"
              value={rawString}
              onChange={e => setRawString(e.target.value)}
            />
          </td>
        </tr>

        <tr>
          <th>
            <label htmlFor="reactive-hash-hashed-value">Hashed by node:crypto :</label>
          </th>
          <td>
            <input
              type="text"
              id="reactive-hash-hashed-value"
              value={hashedString}
              readOnly
            />
          </td>
        </tr>
      </tbody>
    </table>
  );
};

export default ReactiveHash;
