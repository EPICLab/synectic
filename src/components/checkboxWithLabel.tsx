import * as React from 'react';

export class CheckboxWithLabel extends React.Component<{
  labelOn: string;
  labelOff: string;
}, {
  isChecked: boolean;
}> {
  constructor(props: Readonly<{ labelOn: string; labelOff: string }>) {
    super(props);
    this.state = { isChecked: false };
  }

  onChange = (): void => {
    this.setState({ isChecked: !this.state.isChecked });
  }

  render(): JSX.Element {
    return (
      <label>
        <input
          type="checkbox"
          checked={this.state.isChecked}
          onChange={this.onChange}
        />
        {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
      </label>
    );
  }
}