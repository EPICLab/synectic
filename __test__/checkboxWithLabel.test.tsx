import * as React from 'react';
import { shallow } from 'enzyme';
import { CheckboxWithLabel } from '../src/components/checkboxWithLabel';

test('CheckboxWithLabel changes the text after click', () => {
  const checkbox = shallow(<CheckboxWithLabel labelOn="On" labelOff="Off" />);

  // Interaction demo
  expect(checkbox.text()).toEqual('Off');
  checkbox.find('input').simulate('change');
  expect(checkbox.text()).toEqual('On');

  // Snapshot demo
  expect(checkbox).toMatchSnapshot();
});