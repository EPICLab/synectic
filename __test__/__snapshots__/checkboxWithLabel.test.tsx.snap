// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  On
</label>
`;
